src/matching/network.ts

```diff
@@ -11,6 +11,7 @@ function isAnchoredByHostname(
     // Prefix should either be equal to hostname, or end with a full label
     (matchIndex === 0 && (
       hostname.length === filterHostname.length ||
+      filterHostname[filterHostname.length - 1] === '.' ||
```

Comment:
Should we limit to tld suffix, otherwise it means `||foo.baz.` will match `https://foo.baz.foo.bar.com/bar`. Although I'm not sure how likely this will happen

Reply:
True, but I'm pretty sure people rely on this behavior unfortunately. But since we allow both prefix and suffix match of the hostname, some domain could be matched based on its subdomain, although the rule was indented to match another domain. Worst case: we block some more requests!

Reply:
But we could add this check and see if it breaks. My other concern is that it might slow-down matching.